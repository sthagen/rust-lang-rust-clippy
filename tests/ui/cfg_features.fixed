#![warn(clippy::maybe_misused_cfg)]

fn main() {
    #[cfg(feature = "not-really-a-feature")]
    let _ = 1 + 2;

    #[cfg(all(feature = "right", feature = "wrong"))]
    let _ = 1 + 2;

    #[cfg(all(feature = "wrong1", any(feature = "right", feature = "wrong2", feature, features)))]
    let _ = 1 + 2;
}
