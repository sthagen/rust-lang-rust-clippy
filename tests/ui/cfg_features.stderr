error: feature may misspelled as features
  --> $DIR/cfg_features.rs:4:11
   |
LL |     #[cfg(features = "not-really-a-feature")]
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use: `feature = "not-really-a-feature"`
   |
   = note: `-D clippy::maybe-misused-cfg` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::maybe_misused_cfg)]`

error: feature may misspelled as features
  --> $DIR/cfg_features.rs:9:34
   |
LL |     #[cfg(all(feature = "right", features = "wrong"))]
   |                                  ^^^^^^^^^^^^^^^^^^ help: use: `feature = "wrong"`

error: feature may misspelled as features
  --> $DIR/cfg_features.rs:13:15
   |
LL |     #[cfg(all(features = "wrong1", any(feature = "right", features = "wrong2", feature, features)))]
   |               ^^^^^^^^^^^^^^^^^^^ help: use: `feature = "wrong1"`

error: feature may misspelled as features
  --> $DIR/cfg_features.rs:13:59
   |
LL |     #[cfg(all(features = "wrong1", any(feature = "right", features = "wrong2", feature, features)))]
   |                                                           ^^^^^^^^^^^^^^^^^^^ help: use: `feature = "wrong2"`

error: 'test' may be misspelled as 'tests'
  --> $DIR/cfg_features.rs:18:11
   |
LL |     #[cfg(tests)]
   |           ^^^^^ help: do you mean: `test`

error: 'test' may be misspelled as 'Test'
  --> $DIR/cfg_features.rs:21:11
   |
LL |     #[cfg(Test)]
   |           ^^^^ help: do you mean: `test`

error: 'test' may be misspelled as 'tests'
  --> $DIR/cfg_features.rs:25:15
   |
LL |     #[cfg(all(tests, Test))]
   |               ^^^^^ help: do you mean: `test`

error: 'test' may be misspelled as 'Test'
  --> $DIR/cfg_features.rs:25:22
   |
LL |     #[cfg(all(tests, Test))]
   |                      ^^^^ help: do you mean: `test`

error: aborting due to 8 previous errors

